### PR TITLE
OCPBUGS-42301: Switch to github.com/docker/distribution/reference to Mitigate CVE-2024-3727

### PR DESCRIPTION
**Pull request by paul-maidment**
The library github.com/containers/image/v5 has a vulnerability that has as of yet been unresolved.

Thankfully, it is possible to change the part of the library that we use

We can change github.com/containers/image/v5/docker/reference for github.com/docker/distribution/reference

This PR bumps to the latest 4.15 version of the service to pick up the changes already made there

Additionally, minimal dockerfile changes have been made to ensure that we are able to build and test this change locally.

### CODE REVIEW

**Review by gamli75 — APPROVED** [review]

**Comment by paul-maidment** [comment]
@zaneb @andfasano can you help with the required label? thanks

File: Dockerfile.assisted-installer-build
Comment (zaneb) [inline Dockerfile.assisted-installer-build:1]: Only Dockerfile.assisted-installer.ocp is used on this branch, so I wouldn't expect to see this change here. What is it required for?

Comment (paul-maidment) [reply Dockerfile.assisted-installer-build:1]: Agreed, I don't think we need to change this.
I have removed this change from the PR

**Comment by zaneb** [comment]
/label cherry-pick-approved
on behalf of @mhanss

**Review by adriengentil — APPROVED** [review]

**Pull request merged**